modules/charts/src/components/Provider/Colors/colors.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Creates a chart color map for consistent colors across charts
+ * Uses closure-based state, not a React hook
+ *
+ * @param options - Configuration object
+ * @param options.colors - Array of color values to use
+ * @returns Object with methods to manage the color map
+ */
+export const createChartColors = ({ colors }: { colors: Array<string> }) => {
+	const colorMap = new Map<string, string>();
+
+	// used for "color wraparound" modulo
+	let colorIndex = 0;
+
+	return {
+		/**
+		 * Creates a color map for charts
+		 *
+		 * @returns Map of keys to assigned colors
+		 */
+		createColorMap: (data: Map<string, any>) => {
+			if (!data) return;
+			if (colorMap.size !== 0) return;
```

Comment:
is this a good synthesis? not sure yet whether each return is meant to become different per case later. either way, best we keep curlies around conditional blocks, for both consistency and to reduce diff bloat.
```suggestion
			if (!data || colorMap.size !== 0) {
				return;
			}
```